internal/connector/internal/handlers/processors.go

```diff
@@ -0,0 +1,421 @@
+			}
+			originalResource, _ := presenters.PresentProcessor(&dbresource.Processor)
+
+			resource, serr := presenters.PresentProcessor(&dbresource.Processor)
+			if serr != nil {
+				return nil, serr
+			}
+
+			originalSecrets, err := getProcessorSecretRefs(&dbresource.Processor)
+			if err != nil {
+				return nil, errors.GeneralError("could not get existing secrets: %v", err)
+			}
+
+			// Apply the patch..
+			patchBytes, err := io.ReadAll(r.Body)
+			if err != nil {
+				return nil, errors.BadRequest("failed to get patch bytes")
+			}
+
+			// get json resource
+			resourceJson, serr := getResourceJson(resource)
+			if serr != nil {
+				return nil, serr
+			}
+
+			// convert json-patch into merge-patch, since validateProcessorPatch can only validate merge-patch
+			if contentType == JSON_PATCH {
+				patchBytes, err = convertToMergePatch(patchBytes, resourceJson)
+				if err != nil {
+					return nil, errors.BadRequest("failed to convert to merge patch: %v", err)
+				}
+				// changed patch bytes to merge patch
+				contentType = MERGE_PATCH
+			}
+
+			patch := public.ProcessorRequest{}
+			serr = PatchResource(resourceJson, contentType, patchBytes, &patch)
+			if serr != nil {
+				return nil, serr
+			}
+
+			// get and validate patch operation type
+			var operation phase.ProcessorOperation
+			if operation, serr = h.getOperation(resource, patch); err != nil {
+				return nil, serr
+			}
+			if serr = ValidateProcessorOperation(r.Context(), h.namespaceService, &dbresource.Processor, operation,
+				func(processor *dbapi.Processor) *errors.ServiceError {
+					resource.DesiredState = public.ProcessorDesiredState(dbresource.DesiredState)
+					return nil
+				}); serr != nil {
+				return nil, serr
+			}
+
+			// But we don't want to allow the user to update ALL fields... so copy
+			// over the fields that they are allowed to modify...
+			resource.Name = patch.Name
+			resource.Definition = patch.Definition
+			resource.ErrorHandler = patch.ErrorHandler
+			resource.Annotations = patch.Annotations
+			resource.Kafka = patch.Kafka
+			resource.ServiceAccount = patch.ServiceAccount
+
+			// revalidate
+			user := h.authZService.GetValidationUser(r.Context())
+			validates := []handlers.Validate{
+				handlers.Validation("name", &resource.Name, handlers.WithDefault("New Processor"), handlers.MinLen(1), handlers.MaxLen(100)),
+				handlers.Validation("service_account.client_id", &resource.ServiceAccount.ClientId, handlers.MinLen(1)),
+				handlers.Validation("desired_state", (*string)(&resource.DesiredState), handlers.WithDefault("ready"), handlers.IsOneOf(dbapi.ValidProcessorDesiredStates...)),
+				validateProcessorImmutableProperties(patch, originalResource),
+				validatePatchAnnotations(resource.Annotations, originalResource.Annotations),
+				validateProcessor(&resource),
+			}
+
+			// Don't validate user's tenancy in admin api calls
+			// TODO {manstis} Processors are not exposed under the admin API at the moment
```

Comment:
@manstis Out of curiosity, how much work would it be to expose this under the admin API? That would give us the ability to stop/delete Processors that are behaving badly.

Reply:
I'd be tempted to avoid scope creep to be honest.

These PRs are large enough already. Create a JIRA to consider a _consolidated_ Admin API might be a better option.

Reply:
Sure thing 👍

Reply:
See https://issues.redhat.com/browse/MGDCTRS-2291